### PR TITLE
Feature/pe 1089 document terraform provider data sources and resources

### PR DESCRIPTION
**Pull request by kevinkirkup**
Added provider documentation for `data sources` and `resources`.

### CODE REVIEW

**Comment by kevinkirkup** [comment]
@alauer Do you mind reviewing the documentation and providing feedback?
I've added some descriptions but they probably aren't as flushed out as they should be.
There were also some places with `???` where I didn't know what to include.

File: website/docs/r/aws_connection.html.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    * `description` - The description of the cloud gateway.
+
+    * `availability_domain` - The availability domain of the cloud gateway. The valid values are `PRIMARY`, `SECONDARY`.
+
+    * `link_state` - The current link state of the gateway. The valid values are `WAITING_TO_PROVISION`, `PROVISIONING`, `FAILED_TO_PROVISION`, `ACTIVE`, `DOWN`, `UPDATING`, `FAILED_TO_UPDATE`, `DELETING`, `FAILED_TO_DELETE`.
+
+    * `customer_asn` - The customer ASN used for BGP Peering.
+
+    * `customer_ip` - ???
```
Comment (alauer) [inline website/docs/r/aws_connection.html.md]: * `customer_ip` - The assigned IP address to the customer side of the BGP Config.

File: website/docs/r/aws_connection.html.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    * `pureport_asn` - The Pureport ASN used for BGP Peering.
+
+    * `pureport_ip` - ???
+
```
Comment (alauer) [inline website/docs/r/aws_connection.html.md]: * `pureport_ip` - The assigned IP address to the Pureport side of the BGP Config.

File: website/docs/r/aws_connection.html.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    * `bgp_password` - The autogenerated BGP password used for authentication.
+
+    * `peering_subnet` - ???
```
Comment (alauer) [inline website/docs/r/aws_connection.html.md]: * `peering_subnet` - The BGP Config subnet assigned to establish BGP peering.

File: website/docs/r/azure_connection.html.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+    * `description` - The description of the cloud gateway.
+
+    * `availability_domain` - The availability domain of the cloud gateway. The valid values are `PRIMARY`, `SECONDARY`.
+
+    * `link_state` - The current link state of the gateway. The valid values are `WAITING_TO_PROVISION`, `PROVISIONING`, `FAILED_TO_PROVISION`, `ACTIVE`, `DOWN`, `UPDATING`, `FAILED_TO_UPDATE`, `DELETING`, `FAILED_TO_DELETE`.
+
+    * `customer_asn` - The customer ASN used for BGP Peering.
+
+    * `customer_ip` - ???
```
Comment (alauer) [inline website/docs/r/azure_connection.html.md]: * `customer_ip` - The assigned IP address to the customer side of the BGP Config.

File: website/docs/r/azure_connection.html.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+    * `pureport_asn` - The Pureport ASN used for BGP Peering.
+
+    * `pureport_ip` - ???
```
Comment (alauer) [inline website/docs/r/azure_connection.html.md]: * `pureport_ip` - The assigned IP address to the Pureport side of the BGP Config.

File: website/docs/r/azure_connection.html.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    * `bgp_password` - The autogenerated BGP password used for authentication.
+
+    * `peering_subnet` - ???
```
Comment (alauer) [inline website/docs/r/azure_connection.html.md]: * `peering_subnet` - The BGP Config subnet assigned to establish BGP peering.

File: website/docs/r/dummy_connection.html.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+    * `description` - The description of the cloud gateway.
+
+    * `availability_domain` - The availability domain of the cloud gateway. The valid values are `PRIMARY`, `SECONDARY`.
+
+    * `link_state` - The current link state of the gateway. The valid values are `WAITING_TO_PROVISION`, `PROVISIONING`, `FAILED_TO_PROVISION`, `ACTIVE`, `DOWN`, `UPDATING`, `FAILED_TO_UPDATE`, `DELETING`, `FAILED_TO_DELETE`.
+
+    * `customer_asn` - The customer ASN used for BGP Peering.
+
+    * `customer_ip` - ???
```
Comment (alauer) [inline website/docs/r/dummy_connection.html.md]: * `customer_ip` - The assigned IP address to the customer side of the BGP Config.

File: website/docs/r/dummy_connection.html.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+    * `pureport_asn` - The Pureport ASN used for BGP Peering.
+
+    * `pureport_ip` - ???
```
Comment (alauer) [inline website/docs/r/dummy_connection.html.md]: * `pureport_ip` - The assigned IP address to the Pureport side of the BGP Config.

File: website/docs/r/dummy_connection.html.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+    * `bgp_password` - The autogenerated BGP password used for authentication.
+
+    * `peering_subnet` - ???
```
Comment (alauer) [inline website/docs/r/dummy_connection.html.md]: * `peering_subnet` - The BGP Config subnet assigned to establish BGP peering.

File: website/docs/r/site_vpn_connection.html.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+    * `pureport_ip` - ???
+
+    * `bgp_password` - The autogenerated BGP password used for authentication.
+
+    * `peering_subnet` - ???
+
+    * `public_nat_ip` - The public facing IP Address for NAT used by this connection.
+
+    * `customer_gateway_ip` - ???
```
Comment (alauer) [inline website/docs/r/site_vpn_connection.html.md]: * `customer_gateway_ip` - The public IP address of the customers VPN equipment.

File: website/docs/r/site_vpn_connection.html.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+
+    * `customer_vti_ip` - ???
```
Comment (alauer) [inline website/docs/r/site_vpn_connection.html.md]: * `customer_vti_ip` - The assigned IP address to the customer side of the VTI tunnel.

File: website/docs/r/site_vpn_connection.html.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+
+    * `pureport_gateway_ip` - ???
```
Comment (alauer) [inline website/docs/r/site_vpn_connection.html.md]: * `pureport_gateway_ip` - The public IP address of the Pureport VPN gateway.

File: website/docs/r/site_vpn_connection.html.md [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+
+    * `pureport_vti_ip` - ???
```
Comment (alauer) [inline website/docs/r/site_vpn_connection.html.md]: * `pureport_vti_ip` - The assigned IP address to the Pureport side of the VTI tunnel.

**Comment by kevinkirkup** [comment]
@alauer Can you approve this?

**Pull request merged**